scripts/download_osm_data.py

```diff
@@ -105,91 +111,182 @@ def convert_iso_to_geofk(
 
     run = snakemake.config.get("run", {})
     RDIR = run["name"] + "/" if run.get("name") else ""
+    country_list = country_list_to_geofk(snakemake.params.countries)
+
+    # Get unified OSM data configuration
+    osm_config = snakemake.config.get("osm_data", {})
+    source = osm_config.get("source", "download").lower()
 
-    # Check for historical date configuration
-    historical_config = snakemake.config.get("historical_osm_data", {})
-    target_date = historical_config.get("osm_date", None)
+    # Validate source option
+    valid_sources = ["download", "historical", "custom"]
+    if source not in valid_sources:
+        logger.warning(
+            f"Invalid osm_data source '{source}'. Must be one of {valid_sources}. Defaulting to 'download'."
+        )
+        source = "download"
+
+    # Extract configuration based on source
+    custom_path_config = osm_config.get("custom_path", {})
+    target_date = str(osm_config.get("target_date", None))
+
+    # Handle custom_path - dict with pbf/power keys
+    if isinstance(custom_path_config, dict):
+        custom_pbf_path = custom_path_config.get("pbf", "data/custom/osm/pbf")
+        custom_power_path = custom_path_config.get("power", "data/custom/osm/power")
+    else:
+        # Default paths
+        custom_pbf_path = "data/custom/osm/pbf"
+        custom_power_path = "data/custom/osm/power"
 
     # Create date-specific subdirectory for OSM data if historical date is provided
     osm_subdir = ""
-    if target_date:
+    if source == "historical" and target_date:
         if isinstance(target_date, str):
-            osm_subdir = target_date.replace(
-                "-", "")[:6] + "/"  # Format: YYYYMM/
-        else:
+            # Extract YYYYMM from date string
+            osm_subdir = target_date.replace("-", "")[:6] + "/"  # Format: YYYYMM/
+        elif isinstance(target_date, datetime):
             osm_subdir = target_date.strftime("%Y%m") + "/"
+    elif source == "custom":
+        osm_subdir = "custom/"
```

Comment:
As a very minor technical point: is "/" path delimiter needed at the end, given that `osm_subdir ` will be transformed into a `Path` object later?